cpp/include/cuspatial/cubic_spline.hpp

```diff
@@ -84,4 +84,36 @@ std::unique_ptr<cudf::table> cubicspline_coefficients(
   cudf::column_view const& ids,
   cudf::column_view const& offsets,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+namespace detail {
+
+/**
+ * @brief Finds the lower interpolant position of query_points from a set of
+ * interpolation independent variables.
```

Comment:
What does the lower interpolant position mean?  Perhaps more explanation after the @brief.

Reply:
I made another pass at describing this function now in `detail/cubic_spline.hpp` let me know what you think. Otherwise all your other requests are complete.